corelib-definitions/src/main/java/eu/europeana/corelib/definitions/solr/model/Query.java

```diff
@@ -251,20 +255,106 @@ public Query setFacets(List<String> facets) {
 	 * because it is a special query facet
 	 */
 	private void replaceSpecialFacets() {
-		List<String> additionalFacets = new ArrayList<String>();
-		for (String facet : facets) {
-			if (StringUtils.equals("DEFAULT", facet)) {
-				additionalFacets.addAll(new ArrayList<String>(defaultFacets));
-			} else if (StringUtils.equals("REUSABILITY", facet)) {
-				// skip it
-			} else {
-				additionalFacets.add(facet);
-			}
-		}
-		facets = additionalFacets;
+    boolean ok = false;
+		  List<String> additionalFacets = new ArrayList<String>();
+				for (String facet: facets) {
+        if (StringUtils.equalsIgnoreCase("DEFAULT", facet)) {
+            additionalFacets.addAll(defaultFacets);
+        }
+        else if (StringUtils.equalsIgnoreCase("MEDIA", facet)) {
+            additionalFacets.add("has_media");
+        }
+        else if (StringUtils.equalsIgnoreCase("THUMBNAIL", facet)) {
+            additionalFacets.add("has_thumbnails");
+        }
+        else if (StringUtils.equalsIgnoreCase("TEXT_FULLTEXT", facet)) {
+            additionalFacets.add("is_fulltext");
+        }
+        else if (StringUtils.equalsIgnoreCase("REUSABILITY", facet)) {
+            continue;
+        }
+        /*
```

Comment:
Big codeblock of unused code? Why is this not used? Can it be removed?
If you want to keep it in to enable it in a later phase, please add a note to this code block to explain that...